bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java

```diff
@@ -162,12 +164,16 @@ public static void startResourceObservation(Environment env, BString module, BSt
             setObserverContextToCurrentFrame(env, newObserverContext);
 
             newObserverContext.setEntrypointFunctionModule(observerContext.getEntrypointFunctionModule());
-            newObserverContext.setEntrypointFunctionPosition(observerContext.getEntrypointFunctionPosition());
+            newObserverContext.setEntrypointObjectName(observerContext.getEntrypointObjectName());
+            newObserverContext.setEntrypointFunctionName(observerContext.getEntrypointFunctionName());
+            newObserverContext.setEntrypointAccessorName(observerContext.getEntrypointAccessorName());
             newObserverContext.setParent(observerContext);
             observerContext = newObserverContext;
         } else {    // If created now or the listener created to add more tags
             observerContext.setEntrypointFunctionModule(module.getValue());
-            observerContext.setEntrypointFunctionPosition(position.getValue());
+            observerContext.setEntrypointObjectName(serviceName.getValue());
+            observerContext.setEntrypointFunctionName(resourcePathOrFunction.getValue());
+            observerContext.setEntrypointAccessorName(resourceAccessor.getValue());
```

Comment:
We don't need to set accessor for non resource functions.

Reply:
This is for resource observations isn't it?

Reply:
Yes only should be set for resource observability

Reply:
Fixed in cc06a1391b7e2bbe4071119cfa474801cee21e57

Reply:
Fixed in deebe9e0fbd82b278332e377842f2c99023dae07